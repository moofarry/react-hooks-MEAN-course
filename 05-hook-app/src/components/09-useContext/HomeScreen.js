import React from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {

  const { user } = React.useContext(UserContext);

  return (
    <div>
      <h1>HomeScreen</h1>
      <hr />

      <pre>
        {JSON.stringify( user,null,3)}
      </pre>
    </div>
  );
};
