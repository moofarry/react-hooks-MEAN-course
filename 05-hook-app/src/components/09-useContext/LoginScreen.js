import React from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {

  const { setUser } = React.useContext(UserContext);

  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />

      <button className="btn btn-primary" onClick={() => setUser({ id: 123, name: "jhon" })}>
        Login
      </button>
    </div>
  );
};
