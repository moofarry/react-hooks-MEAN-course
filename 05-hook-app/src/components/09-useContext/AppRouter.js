import React from "react";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
  return (
    <Router>
      <NavBar />
      <div className="container">
        <Routes>
          <Route path="/" element={<HomeScreen />} />
          <Route path="/about" element={<AboutScreen />} />
          <Route path="/login" element={<LoginScreen />} />

          <Route path="*" element={<HomeScreen />} />
        </Routes>
      </div>
    </Router>
  );
};
