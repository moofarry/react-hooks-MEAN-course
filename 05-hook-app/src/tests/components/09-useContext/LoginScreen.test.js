import { mount } from "enzyme";

import React from "react";
import { LoginScreen } from "../../../components/09-useContext/LoginScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("test in <LoginScreen/>", () => {
  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  );

  test("should be show correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should be run setUser with args expected", () => {
    wrapper.find("button").prop("onClick")();

    expect(setUser).toHaveBeenCalledWith({ 
      id: 123, name: "jhon"
    });

  });


});
