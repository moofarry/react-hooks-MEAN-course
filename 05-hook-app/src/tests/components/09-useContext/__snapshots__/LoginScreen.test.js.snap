// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <LoginScreen/> should be show correctly 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
